### PR TITLE
v3 - Events + some supporting files 

### PR DESCRIPTION
**Pull request by SharonAllsup**

### CODE REVIEW

File: event.go [hunk 1]
```diff
@@ -0,0 +1,376 @@
+package ari
+
+type Event interface {
+
+	// Get returns a handle pointer to the Event for further interaction
+	Get(name string) *EventHandle
```
Comment (sheenobu) [inline event.go]: The Data and Get methods are unneeded.

File: event.go [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+	Playback Playback `json:"playback"`
+}
+
+/* --------- Recording Events ---------------- */
+
+/* These three are commented out until LiveRecording is taken care of
+// RecordingFailed events indicate that a recording operation request has failed to complete
+type RecordingFailed struct {
+	EventData
+	Recording LiveRecording `json:"recording"`
```
Comment (sheenobu) [inline event.go]: LiveRecording is 'LiveRecordingData' It should compile then.

File: message.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	}
+
+	// Store the raw data
+	e.__raw = &data
+
+	return nil
+}
+
+// AsteriskCode is a websocket Codec for Asterisk messages
+var AsteriskCodec = websocket.Codec{
```
Comment (sheenobu) [inline message.go]: This belongs in client/native, since that is the only part using websockets.

File: event.go [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+	Args []string `json:"args"`
+	ChannelData
+	ReplaceChannel Channel `json:"replacechannel,omitempty"` // TODO: find out what this is
+}
+
+// R2 - SKIPPED FOR NOW
+// GetChannel returns the channel for whom the StasisStart event occurred,
+// optionally attaching a copy of the ARI client
+/*
+func (s *StasisStart) GetChannel(c *Client) Channel {
```
Comment (sheenobu) [inline event.go]: You want to implement this in terms of ari.ChannelHandle and ari.Channel interface.

File: event.go [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+	EventData
+	ChannelData
+}
+
+// ChannelUserevent events are custom-formatted events which have been received
+//   TODO: figure out if these include AMI user events, etc, and how the data is
+//   formatted
+type ChannelUserevent struct {
+	EventData
+	Bridge Bridge `json:"bridge,omitempty"`
```
Comment (sheenobu) [inline event.go]: Bridge -> BridgeData

File: event.go [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+	ChannelData
+	Endpoint  Endpoint    `json:"endpoint,omitempty"`
```
Comment (sheenobu) [inline event.go]: Endpoint -> EndpointData

File: event.go [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+type ChannelHold struct {
+	EventData
+	ChannelData
+	MusicClass string `json:"musicclass"` // The music class being played to the held channel
+}
+
+// ChannelLeftBridge events indicate that a channel has left a bridge
+type ChannelLeftBridge struct {
+	EventData
+	Bridge Bridge `json:"bridge"`
```
Comment (sheenobu) [inline event.go]: Bridge -> BridgeData

File: event.go [hunk 8]
```diff
@@ -0,0 +1,376 @@
...
+	EventData
+	ChannelData
+	Digit       string `json:"digit"`       // The DTMF digit which was received (0-9A-E*#)
+	Duration_ms int    `json:"duration_ms"` // The duration of the DTMF tone, in milliseconds
+}
+
+// ChannelEngteredBridge events indicate that a channel has joined a bridge
+type ChannelEnteredBridge struct {
+	EventData
+	Bridge Bridge `json:"bridge"`
```
Comment (sheenobu) [inline event.go]: Bridge -> BridgeData

File: event.go (hunk 2 above)
Comment (SharonAllsup) [inline event.go]: Fixed in https://github.com/CyCoreSystems/ari/pull/28/commits/6067360c2151c6dfad49c214e58a3bd54b95977d

File: event.go (hunk 5 above)
Comment (SharonAllsup) [inline event.go]: Done in https://github.com/CyCoreSystems/ari/pull/28/commits/6067360c2151c6dfad49c214e58a3bd54b95977d

File: event.go (hunk 6 above)
Comment (SharonAllsup) [inline event.go]: Done in https://github.com/CyCoreSystems/ari/pull/28/commits/6067360c2151c6dfad49c214e58a3bd54b95977d

File: event.go (hunk 7 above)
Comment (SharonAllsup) [inline event.go]: Done in https://github.com/CyCoreSystems/ari/pull/28/commits/6067360c2151c6dfad49c214e58a3bd54b95977d

File: event.go (hunk 8 above)
Comment (SharonAllsup) [inline event.go]: Done in https://github.com/CyCoreSystems/ari/pull/28/commits/6067360c2151c6dfad49c214e58a3bd54b95977d

**Comment by Ulexus** [comment]
With the events already present, is this PR still viable?

**Comment by sheenobu** [comment]
@Ulexus no

**Comment by Ulexus** [comment]
Okay, I'm going to close it, then. @SharonAllsup if we miss something from here, please feel free to open a new request for it.

**Pull request closed without merging**